### PR TITLE
feat(*): reuse preprocessing form fields for config forms and save [khcp-10152]

### PR DESCRIPTION
**Pull request by mptap**
# Summary
https://konghq.atlassian.net/browse/KHCP-10152
Extracted preprocessing form fields logic in reusable helpers, for use in both JSON/YAML config forms and form submission

### CODE REVIEW

File: packages/entities/entities-vaults/src/components/VaultForm.vue [hunk 1]
```diff
@@ -753,69 +753,73 @@ const submitUrl = computed<string>(() => {
   return url
 })
 
-const saveFormData = async (): Promise<void> => {
-  try {
-    form.isReadonly = true
+const getPayload = computed(() => {
```
Comment (jillztom) [inline packages/entities/entities-vaults/src/components/VaultForm.vue:756]: nit: Can you Type this property?

File: packages/entities/entities-plugins/src/components/PluginForm.vue [hunk 2]
```diff
@@ -965,6 +965,23 @@ const isCustomPlugin = computed((): boolean => {
   return !Object.keys(pluginMetaData).includes(props.pluginType)
 })
 
+const getRequestBody = computed(() => {
```
Comment (jillztom) [inline packages/entities/entities-plugins/src/components/PluginForm.vue:968]: nit: Type

File: packages/entities/entities-shared/src/components/common/JsonCodeBlock.vue [hunk 3]
```diff
@@ -57,7 +57,7 @@ const props = defineProps({
   },
 })
 
-const jsonContent = reactive(props.jsonRecord)
+const jsonContent = computed(() => props.jsonRecord)
```
Comment (jillztom) [inline packages/entities/entities-shared/src/components/common/JsonCodeBlock.vue:60]: nit: Type

File: packages/entities/entities-shared/src/components/common/YamlCodeBlock.vue [hunk 4]
```diff
@@ -23,7 +23,7 @@ const props = defineProps({
   },
 })
 
-const yamlContent = reactive(props.yamlRecord)
+const yamlContent = computed(() => props.yamlRecord)
```
Comment (jillztom) [inline packages/entities/entities-shared/src/components/common/YamlCodeBlock.vue:26]: nit: Type

**Comment by jillztom** [comment]
Looks good in general. Added some minor/nit comments

File: packages/entities/entities-plugins/src/components/PluginForm.vue (hunk 2 above)
Comment (mptap) [reply packages/entities/entities-plugins/src/components/PluginForm.vue:968]: Added

File: packages/entities/entities-vaults/src/components/VaultForm.vue (hunk 1 above)
Comment (mptap) [reply packages/entities/entities-vaults/src/components/VaultForm.vue:756]: Added

File: packages/entities/entities-shared/src/components/common/JsonCodeBlock.vue (hunk 3 above)
Comment (mptap) [reply packages/entities/entities-shared/src/components/common/JsonCodeBlock.vue]: Added

File: packages/entities/entities-shared/src/components/common/YamlCodeBlock.vue (hunk 4 above)
Comment (mptap) [reply packages/entities/entities-shared/src/components/common/YamlCodeBlock.vue]: Added

**Review by kaiarrowood — APPROVED** [review]
LGTM! Functionally verified working as expected 🚀

**Pull request merged**